python/cuml/utils/input_utils.py

```diff
@@ -193,9 +194,20 @@ def convert_dtype(X, to_dtype=np.float32):
             return X_m
 
     elif cuda.is_cuda_array(X):
-        X_m = cp.asarray(X)
-        X_m = X_m.astype(to_dtype)
-        return cuda.as_cuda_array(X_m)
+        try:
+            import cupy as cp
```

Comment:
How about doing this at the module top level so it is only done once? I think this is what sklearn tends to do, along with setting a "cupy_installed"-type variable for it. E.g. https://github.com/scikit-learn/scikit-learn/blob/e669a89aa470f3a9e24f344c122f983d26904bb6/sklearn/externals/_pilutil.py

Reply:
Probably not worth it for something so temporary

Reply:
I'm fine pushing as-is and we can always update next week when we know about packaging schedules.